### ISSUE TITLE
Local install of VEP not reliably reading in entire vcf file

### ISSUE THREAD

**Issue by DaRinker** [Ensembl/ensembl-vep#521]
Recently I've been running my local cluster's install of VEP (v94)

For input I've been using simple vcf files that I generate myself.  VEP completes without any errors and the output appears to look normal.

However, today I just happened to notice that some of my output seemed unexpectedly sparse. 
In looking at VEP's summary.html it appears that not all my input variants are reliably read in for processing.  The "Lines of input read" is always one more than "Variants processed".  It's also stopping at the same line on multiple attempts (so it's not random).

If  I go to the lines in the vcf files where VEP stops reading I can find no unexpected or hidden characters or white space. 

Any thoughts as to what's happening? Are there any settings I can use to help troubleshoot?

UPDATE: I am now suspecting that VEP is not playing well with my cluster's job scheduler that will sometimes split the job across two cores of the same node (depending on resources). Am currently re-running same input file on one core.

**Comment by dglemos** [comment]
Hi @DaRinker,
If the problem is not with the cluster, it could be caused by some variant in your input file - could be a variant not supported by vep or malformed that is stopping it unexpectedly. In this case, could you send me your file so that I could try to replicate the issue?

**Comment by DaRinker** [comment]
Hi @dglemos,
Does not seem to be a cluster problem as my single core instance also stopped processing at the same spot. Someone I work with was able to get the full output using their local instal of VEP (same version v94) along with my input.vcf. So it appears that either some of my command line options are causing hte problem and/or that the cluster's installation of VEP has a bug.

FWIW, here's my command line:

 `--af --appris --biotype --cache --check_existing --coding_only --dir_cache /data/my.vep.cache/ --fields Uploaded_variation,Location,Allele,Consequence,IMPACT,SYMBOL,Gene,Feature_type,Feature,BIOTYPE,EXON,INTRON,HGVSc,HGVSp,cDNA_position,CDS_position,Protein_position,Amino_acids,Codons,Existing_variation,DISTANCE,STRAND,FLAGS,SYMBOL_SOURCE,HGNC_ID,TSL,APPRIS,CCDS,ENSP,SWISSPROT,TREMBL,UNIPARC,REFSEQ_MATCH,SOURCE,GIVEN_REF,USED_REF,BAM_EDIT,SIFT,PolyPhen,HGVS_OFFSET,AF,CLIN_SIG,SOMATIC,PHENO,PUBMED --input_file /input.vcf --merged --polyphen b --port 3337 --protein --pubmed --regulatory --sift b --symbol --tab --tsl --uniprot --use_given_ref
--
`

**Comment by dglemos** [comment]
Thanks for the command. I've run your command with a vcf and nothing seems wrong. 
How many variants do you have in the input file and what's the number of `Lines of input read`, `Variants processed` and `Variants filtered out` in the summary output? 
It would be helpful if you could send the line in the input file where vep is stopping.

**Comment by DaRinker** [comment]
The vcf file is >100,000 variants. If I isolate the line in the file where VEP is stopping and make a small vcf of just 11 lines (so the problem line +/- the 5 surrounding lines) everything runs fine.  Conversely, if I remove the problematic line form the large vcf file, then VEP stops on another line. I will send you one of the problematic vcf files offline

**Comment by DaRinker** [comment]
FWIW, here's one of the vcf files that appear to not be completing.

The problem wasn't immediately apparent when I was running whole chromosome
vcf files.  But when I began running whole genome vcf files (like the one
attached) I began noticing that I was missing many chromosomes in the
outputs. Then when I went back I noticed that nearly all of my VEP outputs
were truncated.

**Comment by dglemos** [comment]
I can't see any file attached but you can send your file to [helpdesk](https://www.ensembl.org/Help/Contact).

**Comment by DaRinker** [comment]
I have gotten VEP to complete by using only the default settings.

If I add on the following options I again get truncated output with no errors:
```
--coding_only \
--uniprot \
--symbol \
--protein \
--uniprot \
```
Not sure how much of my experience is relevant to others, but for me it would have been extremely helpful had VEP thrown an alert telling me it didn't make it fully through the input file.

**Comment by dglemos** [comment]
Using `--coding_only` vep only returns consequences that fall in the coding regions of transcripts, in this case it's expected that the output has fewer variants than the input without any warning message. 
I run vep with your command and options and I get the expected results. 
It would be easier if you could give me an exact example of some input variants and the result you are getting.

**Comment by dglemos** [comment]
I'm going to close this issue now. If you have any other questions please feel free to reopen it.

**Comment by frahimov** [comment]
Hello,
I stumbled upon this thread, which seems to be relevant to my problem/question. 
I ran VEP on 9213 variants, however the html output shows 
Lines of input read | 9213
Variants processed | 9365
Lines of input read is correct. 
I am curious where the extra variants indicated in the Variants processed came from.
My script looks like this:
vep \
--gencode_basic \
--input_file inputfile \
--format id \
--cache \
--dir_cache ../cache_dir/ \
--everything \
--output_file outputfile \
--tab

Many thanks!
